python/pyarrow/tests/parquet/conftest.py

```diff
@@ -53,23 +53,22 @@ def s3_bucket(s3_server):
 
 @pytest.fixture
 def s3_example_s3fs(s3_server, s3_bucket):
```

Comment:
Is it useful to keep this, since there's `s3_example_fs` below already?

Reply:
Oh, good point. Will try to unite them and see if I run into issues.

Reply:
I think the point of having both is because we want to test both our own S3FileSystem as also the external s3fs one in some cases ?

Of course now with how you rewrote it, it's duplicated, but don't we have to keep the s3fs one for _some_ tests?

Reply:
Have changed the approach, one more time, to fix the failing tests.

Kept the `s3_example_s3fs ` fixture as was and added a change in `_generate_partition_directories` where the failure was happening due to a separate case of s3fs not being pyarrow `Filesystem`.

[d31a2b2](https://github.com/apache/arrow/pull/39825/commits/d31a2b224d2d40599f8dae864bffd2d904b15432)